test/functional/crud_spec_tests.js

```diff
@@ -127,8 +130,49 @@ describe('CRUD spec', function() {
     });
   });
 
+  function transformBulkWriteResult(result) {
+    const r = {};
+    r.insertedCount = result.nInserted;
+    r.matchedCount = result.nMatched;
+    r.modifiedCount = result.nModified || 0;
+    r.deletedCount = result.nRemoved;
+    r.upsertedCount = result.getUpsertedIds().length;
+    r.upsertedIds = {};
+    r.insertedIds = {};
+
+    // Update the n
+    r.n = r.insertedCount;
+
+    // Inserted documents
+    const inserted = result.getInsertedIds();
+    // Map inserted ids
+    for (let i = 0; i < inserted.length; i++) {
+      r.insertedIds[inserted[i].index] = inserted[i]._id;
+    }
+
+    // Upserted documents
+    const upserted = result.getUpsertedIds();
+    // Map upserted ids
+    for (let i = 0; i < upserted.length; i++) {
+      r.upsertedIds[upserted[i].index] = upserted[i]._id;
+    }
+
+    return r;
+  }
+
   function assertWriteExpectations(collection, outcome) {
     return function(result) {
+      if (outcome.error) {
+        expect(result).to.be.an('error');
+      } else {
+        expect(result).to.not.be.an('error');
+      }
+
+      // TODO: when we fix our bulk write errors, get rid of this
```

Comment:
Is there a ticket for this?

Reply:
I thought there was, b/c it's been a known issue for a while, but I can't seem to find one, so I made https://jira.mongodb.org/browse/NODE-1989